### PR TITLE
Refactor of 'podman prune' to better support remote

### PR DESCRIPTION
**Pull request by jwhonce**
* Push iterations into the service not the client
* Add e2e tests
* Refactor to use new frameworks

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: pkg/adapter/pods.go [hunk 1]
```diff
@@ -4,27 +4,67 @@ package adapter
 
 import (
 	"context"
-	"github.com/pkg/errors"
 	"strings"
 
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
```
Comment (rhatdan) [inline pkg/adapter/pods.go]: Nit: Remove line.

File: pkg/adapter/pods.go [hunk 2]
```diff
@@ -4,27 +4,67 @@ package adapter
...
+	if err != nil {
+		return ok, failures, err
+	}
+	if len(pods) < 1 {
+		return ok, failures, nil
+	}
+
+	pool := shared.NewPool("pod_prune", maxWorkers, len(pods))
+	for _, p := range pods {
+		p := p
```
Comment (rhatdan) [inline pkg/adapter/pods.go]: Why are you doing this?

File: pkg/adapter/runtime.go [hunk 3]
```diff
@@ -7,10 +7,11 @@ import (
 	"context"
 	"io"
 	"io/ioutil"
-	"k8s.io/api/core/v1"
 	"os"
 	"text/template"
 
+	"k8s.io/api/core/v1"
+
```
Comment (rhatdan) [inline pkg/adapter/runtime.go]: Remove empty line.

File: pkg/adapter/runtime.go [hunk 4]
```diff
@@ -43,6 +44,11 @@ type Container struct {
 	*libpod.Container
 }
 
+// Pod ...
```
Comment (rhatdan) [inline pkg/adapter/runtime.go]: Pleas give a description.

File: pkg/adapter/pods.go (hunk 2 above)
Comment (jwhonce) [reply pkg/adapter/pods.go]: Creating copy of pointer keeps GC from de-allocating the pod structure while the pool is being processed.

**Comment by rhatdan** [comment]
LGTM
@mheon @baude @giuseppe @vrothberg @TomSweeneyRedHat PTAL

**Pull request merged**